initial template (#840)
electron/website: blog/electron-37-0.md

The two functions both return a newly-allocated copy of the bitmap and are functionally equivalent.

### Deprecated: Extension methods and events on `session`

`session.loadExtension`, `session.removeExtension`, `session.getExtension`,
`session.getAllExtensions`, and the events `extension-loaded`,
`extension-unloaded`, and `extension-ready` have all moved to the new
[`Extensions` object](https://www.electronjs.org/docs/latest/api/extensions-api)
accessible via the `session.extensions` instance property.

### Removed: `quota` type `syncable` in `session.clearStorageData(options)`

When calling `session.clearStorageData(options)`, the `options.quota` type
`syncable` is no longer supported because it has been
[removed](https://chromium-review.googlesource.com/c/chromium/src/+/6309405)
from upstream Chromium.

### Deprecated: `quota` property in `session.clearStorageData(options)`

When calling `Session.clearStorageData(options)`, the `options.quota`
property is deprecated. Since the `syncable` type was removed, there
is only type left -- `'temporary'` -- so specifying it is unnecessary.

> ### Behavior Changed: `process.exit()` kills utility process synchronously

Author: this point is new

---
title: Electron 37.0.0
date: 2025-06-24T00:00:00.000Z
authors:
  - georgexu99
  - VerteDinde
slug: electron-37-0
tags: [release]
---

Electron 37.0.0 has been released! It includes upgrades to Chromium 138, V8 13.8, and Node 22.16.0.

---

The Electron team is excited to announce the release of Electron 37.0.0! You can install it with npm via `npm install electron@latest` or download it from our [releases website](https://releases.electronjs.org/release?channel=stable). Continue reading for details about this release.

If you have any feedback, please share it with us on [Bluesky](https://bsky.app/profile/electronjs.org) or [Mastodon](https://social.lfx.dev/@electronjs), or join our community [Discord](https://discord.com/invite/electronjs)! Bugs and feature requests can be reported in Electron's [issue tracker](https://github.com/electron/electron/issues).

## Notable Changes

### Corner Smoothing is Here: A Design Enhancement Option

![An image showing different corner smoothing values (0%, 30%, 60%, and 100%) applied to rectangles, with 60% labeled as matching macOS style](../static/assets/img/corner-smoothing.svg)

> Electron 37 introduces a new CSS feature that brings Apple-inspired smooth corners to your applications. The new `-electron-corner-smoothing` CSS property allows for further design customization closer aligning with the MacOS design language. This feature technically landed in Electron 36, but we felt like it deserved a larger spotlight.

Reviewer: ```suggestion
Electron 37 introduces a new CSS feature that brings Apple-inspired smooth corners to your applications. The new `-electron-corner-smoothing` CSS property allows for further design customization closer aligning with the macOS design language. This feature technically landed in Electron 36, but we felt like it deserved a larger spotlight.
```

```css
.my-button {
  border-radius: 48;
  -electron-corner-smoothing: 60%;
}
```

> Unlike the traditional `border-radius` which creates quarter-circle corners, this new property generates a more sophisticated curve that smoothly transitions from the straight edges. The curvature of a squircle's permiter is continuous, whereas a rounded square's is not (Furse, 2018). You can adjust the smoothness from 0% to 100%, or use `system-ui` to automatically match the operating system's style. This subtle but impactful design enhancement affects borders, outlines, and shadows, helping your app feel more native and polished.

Reviewer: We should introduce the `squircle` term in the first paragraph here.

> > **Note:** Read more about [@clavin](https://github.com/clavin)'s squircle [RFC here](https://github.com/electron/rfcs/blob/main/text/0012-corner-smoothing.md). The document goes over guide and reference-level explanations in more detail.

Reviewer: Use `> [!NOTE]` maybe?

https://github.com/orgs/community/discussions/16925

>
> Still wondering what a Squircle is? You can read more about Squircle design in Daniel Furse's blog post, ["Desperately seeking squircles"](https://www.figma.com/blog/desperately-seeking-squircles/).
>
> release notes: Added a CSS rule for smooth corners. [#45185](https://github.com/electron/electron/pull/45185)

##

### Stack Changes

- Chromium `138.0.7204.35`
  - [New in 138](https://developer.chrome.com/blog/new-in-chrome-138/)
  - [New in 137](https://developer.chrome.com/blog/new-in-chrome-137/)

> - Node `22.14.0`

Reviewer: ```suggestion
- Node `22.16.0`
```

version number wasn't updated

> - Node `22.14.0`
>   - [Node 22.14.0 blog post](https://nodejs.org/en/blog/release/v22.16.0/)

Reviewer 2: ```suggestion
- Node `22.16.0`
  - [Node 22.16.0 blog post](https://nodejs.org/en/blog/release/v22.16.0/)
```

- V8 `13.8`

Electron 37 upgrades Chromium from `136.0.7103.48` to `138.0.7204.35`, and V8 from `13.6` to `13.8`.

> ## Electron News

Reviewer: Can we just rename this heading to be GSOC-focused?

> Our Google Summer of Code participants are actively working on their projects! Stay tuned, we have two upcoming projects this Summer.

Reviewer: ```suggestion
Our Google Summer of Code participants are actively working on their projects! Stay tuned, we have two upcoming projects this summer.
```

We're excited about the upcoming [Save/Restore Window State API](https://github.com/electron/rfcs/pull/16/) feature that our Google Summer of Code Partipant, [@nilayarya](https://github.com/nilayarya) is developing.
The new API will provide a built-in, standardized way to handle window state persistence.

> Additionally, [@Hitarth-gg](https://github.com/hitarth-gg) is hard at work restoring and modernizing the Electron Devtron Extension.

Reviewer: ```suggestion
Additionally, [@hitarth-gg](https://github.com/hitarth-gg) is hard at work restoring and modernizing the Electron Devtron Extension.
```

This project will provide tooling for developers to debug IPC communication, track event listeners, and visualize module dependencies in their Electron applications.

> It's an exciting couple of weeks for our GSOC participants, so stay tuned for more updates!

Reviewer: ```suggestion
It has been an exciting couple of weeks for our GSOC participants, so stay tuned for more updates!
```

---
title: Electron 37.0.0
date: 2025-06-24T00:00:00.000Z
authors:
  - georgexu99
  - VerteDinde
slug: electron-37-0
tags: [release]
---

Electron 37.0.0 has been released! It includes upgrades to Chromium 138, V8 13.8, and Node 22.16.0.

---

The Electron team is excited to announce the release of Electron 37.0.0! You can install it with npm via `npm install electron@latest` or download it from our [releases website](https://releases.electronjs.org/release?channel=stable). Continue reading for details about this release.

If you have any feedback, please share it with us on [Bluesky](https://bsky.app/profile/electronjs.org) or [Mastodon](https://social.lfx.dev/@electronjs), or join our community [Discord](https://discord.com/invite/electronjs)! Bugs and feature requests can be reported in Electron's [issue tracker](https://github.com/electron/electron/issues).

## Notable Changes

### Corner Smoothing is Here: A Design Enhancement Option

![An image showing different corner smoothing values (0%, 30%, 60%, and 100%) applied to rectangles, with 60% labeled as matching macOS style](../static/assets/img/corner-smoothing.svg)

> Electron 37 (ok, you got us! This feature technically landed in Electron 36, but we felt like it deserved a larger spotlight) introduces a new CSS feature that brings Apple-inspired smooth corners to your applications. The new `-electron-corner-smoothing` CSS property allows for further design customization closer aligning with the MacOS design language.

Reviewer 3: I think I'd put the "ok, you got us" into a separate line (or omit it). It kind of breaks the flow.

Reviewer 3: You could also say: "We backported this feature to Electron 36 to let you use it earlier."

We're excited about the upcoming [Save/Restore Window State API](https://github.com/electron/rfcs/pull/16/) feature that our Google Summer of Code Partipant, [@nilayarya](https://github.com/nilayarya) is developing.
The new API will provide a built-in, standardized way to handle window state persistence.

Additionally, [@Hitarth-gg](https://github.com/hitarth-gg) is hard at work restoring and modernizing the Electron Devtron Extension.
This project will provide tooling for developers to debug IPC communication, track event listeners, and visualize module dependencies in their Electron applications.

It's an exciting couple of weeks for our GSOC participants, so stay tuned for more updates!

### New Features and Improvements

- Added `innerWidth` and `innerHeight` options for `window.open`. [#47039](https://github.com/electron/electron/pull/47039) (Also in [35](https://github.com/electron/electron/pull/47045), [36](https://github.com/electron/electron/pull/47038))
- Added `before-mouse-event` to allow intercepting and preventing mouse events in `webContents`. [#47364](https://github.com/electron/electron/pull/47364) (Also in [36](https://github.com/electron/electron/pull/47365))

> - Added `nativeTheme.shouldUseDarkColorsForSystemIntegratedUI` to distinguish system and app theme. [#46438](https://github.com/electron/electron/pull/46438) Also in [35](https://github.com/electron/electron/pull/46599), [36](https://github.com/electron/electron/pull/46598)

Reviewer 3: ```suggestion
```

This was already part of the v36 blog post.

- Added support for `--no-experimental-global-navigator` flag. [#47418](https://github.com/electron/electron/pull/47418) (Also in [35](https://github.com/electron/electron/pull/47416), [36](https://github.com/electron/electron/pull/47417))
- Added support for `screen.dipToScreenPoint(point)` and `screen.screenToDipPoint(point)` on Linux X11. [#46895](https://github.com/electron/electron/pull/46895) (Also in [35](https://github.com/electron/electron/pull/47124), [36](https://github.com/electron/electron/pull/47125))
- Added support for menu item role `palette` and `header` on macOS. [#47245](https://github.com/electron/electron/pull/47245)
- Added support for node option `--experimental-network-inspection`. [#47031](https://github.com/electron/electron/pull/47031) (Also in [35](https://github.com/electron/electron/pull/47029), [36](https://github.com/electron/electron/pull/47030))
- Exposed `win.isContentProtected()` to allow developers to check window protection status. [#47310](https://github.com/electron/electron/pull/47310) (Also in [36](https://github.com/electron/electron/pull/47311))

> - Improved performance of `desktopCapturer.getSources` when not requesting thumbnails on macOS. [#46138](https://github.com/electron/electron/pull/46138) (Also in [34](https://github.com/electron/electron/pull/46250), [35](https://github.com/electron/electron/pull/46249), [36](https://github.com/electron/electron/pull/46251))

Reviewer 3: ```suggestion
```

This was already part of the v36 blog post.

Electron 37 introduces a new CSS feature that brings Apple-inspired smooth corners to your applications. The new `-electron-corner-smoothing` CSS property allows for further design customization closer aligning with the MacOS design language.

```css
.my-button {
  border-radius: 48;
  -electron-corner-smoothing: 60%;
}
```

Unlike traditional `border-radius` which creates circular corners, this new property generates a more sophisticated curve that smoothly transitions from the straight edges. You can adjust the smoothness from 0% to 100%, or use `system-ui` to automatically match the operating system's style. This subtle but impactful design enhancement affects borders, outlines, and shadows, helping your app feel more native and polished.

> **Note:** Read more about [@clavin](https://github.com/clavin)'s squircle [RFC here](https://github.com/electron/rfcs/blob/main/text/0012-corner-smoothing.md). The document goes over guide and reference-level explanations in more detail.
>
> release notes: Added a CSS rule for smooth corners. [#45185](https://github.com/electron/electron/pull/45185)

### New Window Management Improvements

> - Added `BrowserWindow.isSnapped()` to indicate whether a given window has been arranged via Snap. [#46079](https://github.com/electron/electron/pull/46079) (Also in [36](https://github.com/electron/electron/pull/46226))

Reviewer 3: ```suggestion
```

This was already part of the v36 blog post.

> **Note:** Stay tuned! We're excited about the upcoming [Save/Restore Window State API](https://github.com/electron/electron/pull/47425) feature that our Google Summer of Code Partipant, [@nilayarya](https://github.com/nilayarya) is developing.
>
> The new API will provide a built-in, standardized way to handle window state persistence.

### Additional GSOC Plug:

// TODO: add plug for Hitarth's Devtron changes

##

### Stack Changes

- Chromium `138.0.7204.35`
  - [New in 138](https://developer.chrome.com/blog/new-in-chrome-138/)
  - [New in 137](https://developer.chrome.com/blog/new-in-chrome-137/)
- Node `22.14.0`
  - [Node 22.14.0 blog post](https://nodejs.org/en/blog/release/v22.16.0/)
- V8 `13.8`

Electron 37 upgrades Chromium from `136.0.7103.48` to `138.0.7204.35`, and V8 from `13.6` to `13.8`.

### New Features and Improvements

- Added `before-mouse-event` to allow intercepting and preventing mouse events in WebContents. [#47364](https://github.com/electron/electron/pull/47364) (Also in [36](https://github.com/electron/electron/pull/47365))

> - Added `ffmpeg.dll` to delay load configuration. [#46151](https://github.com/electron/electron/pull/46151) (Also in [34](https://github.com/electron/electron/pull/46174), [35](https://github.com/electron/electron/pull/46172), [36](https://github.com/electron/electron/pull/46173))

Reviewer 3: ```suggestion
```

This was already part of the v36 blog post.

- Added `nativeTheme.shouldUseDarkColorsForSystemIntegratedUI` to distinguish system and app theme. [#46438](https://github.com/electron/electron/pull/46438) Also in [35](https://github.com/electron/electron/pull/46599), [36](https://github.com/electron/electron/pull/46598)
- Added `scriptURL` property to `ServiceWorkerMain`. [#45863](https://github.com/electron/electron/pull/45863)

> - Added sublabel functionality for menus on macOS >= 14.4. [#47042](https://github.com/electron/electron/pull/47042) (Also in [35](https://github.com/electron/electron/pull/47041), [36](https://github.com/electron/electron/pull/47040))

Reviewer 3: ```suggestion
- Added `sublabel` functionality for menus on macOS >= 14.4. [#47042](https://github.com/electron/electron/pull/47042) (Also in [35](https://github.com/electron/electron/pull/47041), [36](https://github.com/electron/electron/pull/47040))
```

> - Added support for Autofill, Writing Tools and Services macOS level menu items in context menus via the new `frame` option in `menu.popup`. [#45138](https://github.com/electron/electron/pull/45138) (Also in [36](https://github.com/electron/electron/pull/46350))

Reviewer 3: ```suggestion
```

This was already part of the v36 blog post.

- Added support for `--no-experimental-global-navigator` flag. [#47418](https://github.com/electron/electron/pull/47418) (Also in [35](https://github.com/electron/electron/pull/47416), [36](https://github.com/electron/electron/pull/47417))
- Added support for `screen.dipToScreenPoint(point)` and `screen.screenToDipPoint(point)` on Linux X11. [#46895](https://github.com/electron/electron/pull/46895) (Also in [35](https://github.com/electron/electron/pull/47124), [36](https://github.com/electron/electron/pull/47125))
- Added support for `system-context-menu` on Linux. [#45848](https://github.com/electron/electron/pull/45848) (Also in [35](https://github.com/electron/electron/pull/46977), [36](https://github.com/electron/electron/pull/46399))
- Added support for menu item role `palette` and `header` on macOS. [#47245](https://github.com/electron/electron/pull/47245)
- Added support for node option `--experimental-network-inspection`. [#47031](https://github.com/electron/electron/pull/47031) (Also in [35](https://github.com/electron/electron/pull/47029), [36](https://github.com/electron/electron/pull/47030))

> - Added the priority and priorityIncremental options to net.request(). [#47321](https://github.com/electron/electron/pull/47321) (Also in [36](https://github.com/electron/electron/pull/47320))

Reviewer 3: ```suggestion
- Added the `priority` and `priorityIncremental` options to `net.request()`. [#47321](https://github.com/electron/electron/pull/47321) (Also in [36](https://github.com/electron/electron/pull/47320))
```

- Exposed `win.isContentProtected()` to allow developers to check window protection status. [#47310](https://github.com/electron/electron/pull/47310) (Also in [36](https://github.com/electron/electron/pull/47311))

> - Improved ASAR integrity checks on Windows. [#46509](https://github.com/electron/electron/pull/46509) (Also in [36](https://github.com/electron/electron/pull/46537))

Reviewer 3: ```suggestion
```

This was already part of the v36 blog post.

> - Improved performance of desktopCapturer.getSources when not requesting thumbnails on macOS. [#46138](https://github.com/electron/electron/pull/46138) (Also in [34](https://github.com/electron/electron/pull/46250), [35](https://github.com/electron/electron/pull/46249), [36](https://github.com/electron/electron/pull/46251))

Reviewer 3: ```suggestion
- Improved performance of `desktopCapturer.getSources` when not requesting thumbnails on macOS. [#46138](https://github.com/electron/electron/pull/46138) (Also in [34](https://github.com/electron/electron/pull/46250), [35](https://github.com/electron/electron/pull/46249), [36](https://github.com/electron/electron/pull/46251))
```
